benchexec/tools/esbmc.py

```diff
@@ -69,3 +69,22 @@ def determine_result(self, run):
                 status = "ERROR"
 
         return status
+    
+    def get_value_from_output(self, output, identifier):
+        regex = re.compile(identifier)
+        matches = list()
+
+        # Match first element of each line
+        for line in output:
+            match = regex.search(line.strip())
+            if match and len(match.groups()) >= 1:
+                matches.append(match.group(1))
+
+        if len(matches) == 1:
+            return matches[0]
+        elif len(matches) > 1:
+            total = 0
+            for x in matches:
+                total = total + float(x)
```

Comment:
If you care a lot about precision, you could use the `decimal` package here, which implements arbitrary-precision decimals. And you can keep the value as is when returning without `str()`, BenchExec will handle this appropriately.

The whole `elif` body part could also be replaced with
`return sum(float(value) for value in matches)`
(equivalently with `decimal.Decimal(value)` instead of `float`.

Reply:
I did the change to use `decimal.Decimal`. However, if I don't do the cast, I will get `AttributeError: 'decimal.Decimal' object has no attribute 'lower'` at `benchexec/tablegenerator/util.py", line 157` at the end.

Reply:
Sorry, I had misremembered. But it is a nice feature to have, so I added it in e61f7b103365f0f9d9b36aca619394bc5cdb0bbf and changed your tool-info module accordingly.